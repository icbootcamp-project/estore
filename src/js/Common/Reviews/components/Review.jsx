import React  from 'react';
import { string,number } from "prop-types";

function renderRating(rat){
    const stars = [];
    
    for( let i =1; i <= rat; i += 1 ){
        stars.push(
            <img src="./img/products-page-details/star_rating.png" height="25" width="25" alt="Rating" />
        );
    }

    return stars;
}

const Review =(props) => (
    <div className="review-content">
        <h2>{props.name}</h2>
        <p>
            {props.comment}
        </p>
        <span>{props.time}</span>
        <div className="rating">
        { renderRating(props.rating) }
        </div>
    </div>

    );
Review.propTypes = {
    name:string.isRequired,
    comment:string.isRequired,
    rating:number.isRequired,
    time:string.isRequired

}
export default Review;