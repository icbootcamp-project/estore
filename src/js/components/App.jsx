import React from 'react';
import { connect } from 'react-redux';
import { func, shape, arrayOf, string, bool, any } from 'prop-types';
import {
  BrowserRouter as Router,
  Route,
  Link,
  Switch,
  withRouter
} from 'react-router-dom';
import * as actions from '../actions';
import Dashboard from './Dashboard/Dashboard';
import Home from './Dashboard/Home';
import Form from './Form';
import Students from './Students';
import { isValidToken } from '../utils';
import axios from 'axios';

export class App extends React.Component {
  static propTypes = {
    fetchData: func.isRequired,
    data: arrayOf(any).isRequired,
    isLoading: bool.isRequired,
    error: shape({
      message: string
    })
  };

  static defaultProps = {
    error: null
  };

  state = {
    isAuthenticated: false
  };

  componentDidMount() {
    const { fetchData } = this.props;
    fetchData();
    const token = localStorage.getItem('token');
    console.log('token', token);

    if (isValidToken()) {
      console.log('token');
      this.setState({ isAuthenticated: true });
    } else {
      this.props.history.push('/form');
    }
    const testData = {
      name: 'foodx',
      imgUrl: 'http://www.estore.com/img/img2.jpg',
      subCategorys: ['dry fruits', 'hot food']
    };
    axios.post('/category', testData).then(category => {
      console.log('created category is ', category.data);
    });
  }

  render() {
    console.log('props are', this.props);
    const { data, isLoading, error } = this.props;
    const { isAuthenticated } = this.state;

    return (
      <div className="app">
        {`data retreived is ${JSON.stringify(data)}, ${isLoading}, ${error &&
          error.message}`}
        <br />
        <Link style={{ margin: '20px' }} to="/dashboard">
          Go to Dashboard
        </Link>
        {isAuthenticated && <Link to="/dashboard/profile">Go to Profile</Link>}
        <Link to="/form">Go to Form</Link>
        <br />
        {isAuthenticated && <Link to="/students">Students</Link>}

        <Switch>
          <Route path="/dashboard" component={Dashboard} />
          <Route path="/form" component={Form} />
          <Route path="/students" component={Students} />
        </Switch>
      </div>
    );
  }
}

function mapStateToProps({ data }) {
  return {
    data: data.data,
    isLoading: data.isLoading,
    error: data.error
  };
}

export default withRouter(
  connect(mapStateToProps, { fetchData: actions.fetchData })(App)
);
