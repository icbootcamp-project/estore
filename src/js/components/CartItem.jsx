// ########## Import Dependencies Here ##########
import React, { Fragment } from "react";
import { shape, func } from "prop-types";

// ########## Import Components Here ##########
import Counter from "./Counter";

const CartItem = props => {
  const { cartItem, addCounter, lessCounter } = props;
  return (
    <Fragment>
      <div className="cart-item">
        <div className="img">
          <div className={`item-frame-${cartItem.cartItemId.slice(5)}`} />
        </div>

        <div className="detail">
          <h3 className="item-name">{cartItem.cartItemName}</h3>
          <p className="item-size">{cartItem.cartItemSize} </p>
          <div className="co">
            <div id="counter">
              <Counter
                addCounter={addCounter}
                lessCounter={lessCounter}
                cartItem={cartItem}
              />
            </div>
            <div className="del-icon" />
          </div>
        </div>

        <div className="sec-amt">
          <div className="amount">Rs: {cartItem.cartItemDisplayPrice}</div>
        </div>
      </div>
    </Fragment>
  );
};

CartItem.propTypes = {
  cartItem: shape().isRequired,
  addCounter: func.isRequired,
  lessCounter: func.isRequired
};

export default CartItem;
