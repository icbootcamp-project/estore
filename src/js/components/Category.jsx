// ########## Import Dependencies Here ##########
import React, { Fragment } from "react";
import { shape, string } from "prop-types";

// ########## Import Components Here ##########

const Category = props => {
  const { category } = props;
  return (
    <Fragment>
      <div className="category-name">{category.name.toUpperCase()}</div>
    </Fragment>
  );
};

Category.propTypes = {
  category: shape({
    name: string.isRequired
  }).isRequired
};

export default Category;
