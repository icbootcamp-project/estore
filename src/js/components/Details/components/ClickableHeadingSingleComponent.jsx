import React from "react";

import { bool, shape, func } from "prop-types";

const ClickableHeadingSingleComponent = ({ item, isActive, clickFunction }) => (
  <div
    className="clickable-headings-div"
    style={
      //   this.state[
      //   Object.keys(this.state)[clickableArray.indexOf(item)]
      // ]

      (isActive && {
        color: "blue",
        borderStyle: "solid",
        borderWidth: "0 0 5px 0",
        borderColor: "blue"
      }) ||
      {}
    }
    // key={item.text}
    onClick={clickFunction}
    // onKeyPress={clickableFunc[clickableArray.indexOf(item)]}
    role="none"
  >
    {item.text}
  </div>
);

ClickableHeadingSingleComponent.propTypes = {
  isActive: bool.isRequired,
  item: shape().isRequired,
  clickFunction: func.isRequired
};

export default ClickableHeadingSingleComponent;
