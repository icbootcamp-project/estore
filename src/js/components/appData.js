const header = {
  icons: [
    {
      iconId: "header-icon-1",
      iconName: "search icon",
      iconPosition: "far-left",
      iconPath: "./img/header/search-icon.png"
    },
    {
      iconId: "header-icon-2",
      iconName: "favourite icon",
      iconPosition: "left",
      iconPath: "./img/header/favourite-icon.png"
    },
    {
      iconId: "header-icon-3",
      iconName: "cart icon",
      iconPosition: "right",
      iconPath: "./img/header/cart-icon.png"
    },
    {
      iconId: "header-icon-4",
      iconName: "menu icon",
      iconPosition: "far-right",
      iconPath: "./img/header/res-menu-icon.png"
    }
  ],
  logo: {
    logoPath: "./img/header/estore.png"
  }
};

const categories = [
  {
    id: "category-1",
    name: "grocery"
  },
  {
    id: "category-2",
    name: "food"
  },
  {
    id: "category-3",
    name: "house care"
  },
  {
    id: "category-4",
    name: "sale"
  }
];

const footer = [];

const appData = [header, categories, footer];

export default appData;
