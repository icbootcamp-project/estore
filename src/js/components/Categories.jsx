// ########## Import Dependencies Here ##########
import React, { Fragment } from "react";
import { arrayOf, shape } from "prop-types";

// ########## Import Components Here ##########
import Category from "./Category";

const Categories = props => {
  function renderCategories() {
    const { categories } = props;
    return categories.map(item => <Category key={item.id} category={item} />);
  }

  return (
    <Fragment>
      <div id="categories">
        <div className="category-item">
          <img
            className="left-arrow"
            src="/img/categories/arrow.png"
            alt="left arrow"
          />
          {renderCategories()}
          <img
            className="right-arrow"
            src="/img/categories/arrow.png"
            alt="left arrow"
          />
        </div>
      </div>
    </Fragment>
  );
};

Categories.propTypes = {
  categories: arrayOf(shape()).isRequired
};

export default Categories;
