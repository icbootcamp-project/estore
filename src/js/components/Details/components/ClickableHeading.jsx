import React, { Fragment } from "react";

import PriceButton from "./PriceButton";

import VariationDetailsPage from "./VariationDetailsPage";

import ClickableHeadingSingleComponent from "./ClickableHeadingSingleComponent";
import array from "./detailsAppData";

export default class ClickableHeading extends React.Component {
  constructor() {
    super();
    this.state = { data: array, goto: "Details" };
  }
  // func1 = () =>
  //   this.setState({
  //     value: { text: "Details"}
  //   });

  clickIt = i => {
    const { data } = this.state;
    // console.log("number of this element is ", i);
    // const changedData = data.map((item, index) => {
    //   item.isActive = index === i;
    //   // console.log("true is a ", a.isActive);
    //   return item;
    // });
    // const divToRender = "";
    const changedData = data.map((item, index) => {
      if (index === i) {
        item.isActive = true;
        this.setState({
          goto: item.text
        });
      } else {
        item.isActive = false;
      }
      return item;
    });

    // console.log("using for each it gives an array  ", changedData);
    this.setState({
      data: changedData
    });
  };

  // func2 = () =>
  //   this.setState({
  //     value: { text: "Reviews", isActive: true }
  //   });
  // color: $primary-color;
  // border-style: solid;
  // border-width: 0 0 5px 0;
  // border-color: $primary-color;
  render() {
    // const clickableArray = array;
    // console.log(array);
    // console.log("asdfasssssssssssssssssssssssssssssssssss", this.state.value);
    // const clickableFunc = [this.func1, this.func2];
    const mapDataToRender = this.state.data.map((item, index) => (
      // if (item.isActive) {
      //   console.log("is active for this component is ", item.isActive);
      // }

      <ClickableHeadingSingleComponent
        item={item}
        isActive={item.isActive}
        key={item.text}
        // clickFunction={clickableFunc[index]}
        clickFunction={() => this.clickIt(index)}
      />
    ));
    return (
      <Fragment>
        <div className="clickable-headings-outer-div">{mapDataToRender}</div>
        {(this.state.goto === "Details" && (
          <Fragment>
            <div className="clickable-description-div">
              description div is present here and it will be routed either to
              details or to reviews according to situation or according to the
              click
            </div>
            {/* THIS IS VARIATION COMPONENT WHERE OPTIONS CAN BE SELECTED */}
            <div className="to-display-two-things border-line-between-options">
              <VariationDetailsPage {...this.props} text="Size:" />
              <VariationDetailsPage {...this.props} text="Shape:" />
            </div>

            {/* HERE border-line-between-options class IS COPED WITH THE border1 id IN VariationDetailsPage TO
              GET THE FUNCTIONALITY OF THE LINE BETWEEN THE VARIATION OPTIONS AND ITS CSS IS WRITTEN IN
               variationOptions.scss */}
            <div className="to-display-two-things border-line-between-options">
              <VariationDetailsPage {...this.props} text="Variation:" />
              <VariationDetailsPage {...this.props} text="More varitation:" />
            </div>
            {/* THIS PORTION IS FOR THE LOWER BODY OF DETAILS PAGE  */}
            <div className="full-outer-div">
              <div className="outer-details-div">
                <div className="details">
                  <PriceButton
                    color="#f5a219"
                    isOpen
                    isImage
                    value="PKR: 250 Rs"
                  />
                </div>
                <div className="details">
                  <PriceButton
                    color="#c52f2f"
                    isOpen
                    isImage
                    value="PKR: 250 Rs"
                  />
                </div>
                <div className="details">
                  <PriceButton
                    color="#45c34d"
                    isOpen
                    isImage
                    value="PKR: 250 Rs"
                  />
                </div>
                {/* <Payment /> */}
              </div>
              <div className="details">
                <PriceButton color="#f5de46" isOpen value="Add to Cart" />
              </div>
              <div className="clear" />
            </div>
          </Fragment>
        )) || (
          <div className="clickable-description-div">
            review div
            {/*
          YOU HAVE TO WRITE THE WHOLE PORTION OF REVIEW_PORTION OF THE DETAILS PAGE IN HERE */}
          </div>
        )}
      </Fragment>
    );
  }
}

// const clickableArray = ["Details", "Review"];
// const mapDataToRender = clickableArray.map(item => (
//   <div className="clickable-headings-div" key={clickableArray.indexOf(item)}>
//     {item}
//   </div>
// ));
// const ClickableHeading = () => (
//   <Fragment>
//     <div className="clickable-headings-outer-div">{mapDataToRender}</div>
//     {this.state.detailsIsOpen ? <div className="clickable-description-div">
//       description div is present here and it will be routed either to details or
//       to reviews according to situation or according to the click
//     </div>}
//   </Fragment>
// );
