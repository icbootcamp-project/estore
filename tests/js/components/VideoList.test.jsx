import React from 'react';
import { VideoList } from '../../../src/js/components/VideoList';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('VideoList Component', () => {
  let videoList, selectVideo;

  beforeEach(() => {
    videoList = [
      {
        id: { videoId: '1' },
        snippet: {
          title: 'some title',
          description: 'some description',
          thumbnails: {
            default: {
              url: 'someurl'
            }
          }
        }
      },
      {
        id: { videoId: '2' },
        snippet: {
          title: 'some title2',
          description: 'some description2',
          thumbnails: {
            default: {
              url: 'someurl2'
            }
          }
        }
      }
    ];
    selectVideo = jest.fn();
  });

  it('renders', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Loading message', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={true}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Error message', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={{ message: 'Error While loading Videos' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('passes on the selectVideo fn', () => {
    expect(selectVideo.mock.calls.length).toBe(0);
  });
});
