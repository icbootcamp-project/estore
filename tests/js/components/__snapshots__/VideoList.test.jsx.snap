// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList Component renders 1`] = `
<div
  className="videoList"
>
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="1"
    video={
      Object {
        "id": Object {
          "videoId": "1",
        },
        "snippet": Object {
          "description": "some description",
          "thumbnails": Object {
            "default": Object {
              "url": "someurl",
            },
          },
          "title": "some title",
        },
      }
    }
  />
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="2"
    video={
      Object {
        "id": Object {
          "videoId": "2",
        },
        "snippet": Object {
          "description": "some description2",
          "thumbnails": Object {
            "default": Object {
              "url": "someurl2",
            },
          },
          "title": "some title2",
        },
      }
    }
  />
</div>
`;

exports[`VideoList Component renders Error message 1`] = `
<div
  className="videoList"
>
  <div>
    Error While loading Videos
  </div>
</div>
`;

exports[`VideoList Component renders Loading message 1`] = `
<div
  className="videoList"
>
  <div>
    Loading Videos.....
  </div>
</div>
`;
